src/Streamly/Internal/System/Process.hs

```diff
@@ -357,25 +358,25 @@ createProc' modCfg path args = do
     Config cfg = modCfg $ mkConfig path args
 
 {-# INLINE putChunksClose #-}
-putChunksClose :: (MonadIO m, IsStream t) =>
-    Handle -> t m (Array Word8) -> t m a
+putChunksClose :: MonadIO m =>
+    Handle -> Stream m (Array Word8) -> Stream m a
 putChunksClose h input =
     Stream.before
-        (Handle.putChunks h (adapt input) >> liftIO (hClose h))
+        (Handle.putChunks h input >> liftIO (hClose h))
         Stream.nil
 
 {-# INLINE toChunksClose #-}
-toChunksClose :: (IsStream t, MonadAsync m) => Handle -> t m (Array Word8)
+toChunksClose :: MonadAsync m => Handle -> Stream m (Array Word8)
```

Comment:
Maybe we can do with a simpler constraint here, without MonadThrow? Does `after` require a MonadThrow?

Reply:
Yes, `after` requires `MonadAsync`.